src/backend/postmaster/postmaster.c

```diff
@@ -1963,6 +1970,39 @@ ServerLoop(void)
 			}
 		}
 
+#ifdef FAULT_INJECTOR
+		if (pmState == PM_RUN)
+		{
+			/*
+			 * Code below is for test purposes only. If a fault is set, it
+			 * will delay the termination of the background writer process for
+			 * the amount of seconds desired by the test.
+			 *
+			 * Important note: it is NOT allowed to access the fault injector
+			 * from Postmaster process or from any SIGQUIT handler of child
+			 * processes when the system is resetting after the crash of some
+			 * backend. Reason: during reset Postmaster kills child processes,
+			 * and it might kill some process when the process has acquired
+			 * the spin lock of the fault injector, but hasn't released it
+			 * yet. So, subsequent call of the fault injector api from
+			 * Postmaster or any SIGQUIT handler will lead to deadlock. Only
+			 * 'doomed' processes can still call the fault injector api, as
+			 * they will soon be terminated anyway. Thus, the fault injector
+			 * is accessed only when the Postmaster state is PM_RUN.
+			 */
+			if (SIMPLE_FAULT_INJECTOR_WITH_PARAM("postmaster_delay_termination_bg_writer", \
+										 &delay_bg_writer_termination_sec) ==
```

Comment:
Maybe this indentation will be better:
```suggestion
			if (SIMPLE_FAULT_INJECTOR_WITH_PARAM("postmaster_delay_termination_bg_writer",
												 &delay_bg_writer_termination_sec) == FaultInjectorTypeSkip)
```

Reply:
> Maybe this indentation will be better:

Updated, but put the indentation provided by 'pgindent'